pkg/controller/cloudprivateipconfig/cloudprivateipconfig_controller.go

```diff
@@ -264,7 +264,11 @@ func (c *CloudPrivateIPConfigController) SyncHandler(key string) error {
 		klog.Infof("CloudPrivateIPConfig: %q will be deleted from node: %q", key, nodeNameToDel)
 
 		node, err := c.nodesLister.Get(nodeNameToDel)
-		if err != nil {
+		// there is a case when the node was deleted and the ip still needs to be released. if the node
+		// doesn't exist, nodesLister.Get() will still return and error but we don't want to return with it
+		if node == nil {
```

Comment:
This comparison is not correct.
If err is set it is very likely that node will be set to nil, it would make us always enter the first if.
Instead please add an additional check under `if err != nil` that checks the error itself:
```
if err != nil {
    if apierrors.IsNotFound(err) {
        klog.Infof("Node: %s no longer exists. Will still need to unassign CloudPrivateIPConfig: %q", nodeNameToDel, key)
    } else {
        return err
    }
}

```